tests/Shared/WorkloadTesting/BuildEnvironment.cs

```diff
@@ -166,19 +168,66 @@ public BuildEnvironment(bool useSystemDotNet = false, TestTargetFramework target
             LogRootPath = Path.Combine(AppContext.BaseDirectory, "logs");
         }
 
-        if (Directory.Exists(TestRootPath))
+        Console.WriteLine($"*** [{TargetFramework}] Using path for projects: {TestRootPath}");
+        CleanupTestRootPath();
+        if (!Directory.Exists(TestRootPath))
         {
-            Directory.Delete(TestRootPath, recursive: true);
+            Directory.CreateDirectory(TestRootPath);
         }
 
-        Directory.CreateDirectory(TestRootPath);
-
         Console.WriteLine($"*** [{TargetFramework}] Using workload path: {sdkForWorkloadPath}");
         if (HasWorkloadFromArtifacts)
         {
-            Console.WriteLine($"*** [{TargetFramework}] Using NuGet cache (never deleted automatically): {NuGetPackagesPath}");
+            if (EnvironmentVariables.IsRunningOnCI)
+            {
+                Console.WriteLine($"*** [{TargetFramework}] Using NuGet cache: {NuGetPackagesPath}");
+                if (Directory.Exists(NuGetPackagesPath))
+                {
+                    Directory.Delete(NuGetPackagesPath, recursive: true);
+                }
+            }
+            else
+            {
+                Console.WriteLine($"*** [{TargetFramework}] Using NuGet cache (never deleted automatically): {NuGetPackagesPath}");
+            }
+        }
+
+        static void CleanupTestRootPath()
+        {
+            if (!Directory.Exists(TestRootPath))
+            {
+                return;
+            }
+
+            try
+            {
+                Directory.Delete(TestRootPath, recursive: true);
+            }
+            catch (IOException)
+            {
+                if (EnvironmentVariables.IsRunningOnCI)
+                {
+                    // Investigate failures on CI, so don't guard against exceptions
```

Comment:
What does this comment mean?

Reply:
Basically that we want to fail on CI if deleting the `testroot` throws an exception, as it would be something that should be checked manually, like some lingering process from a previous run holding onto a file handle in `testroot`.

Reply:
>  that we want to fail on CI if deleting the testroot throws an exception

How does swallowing the exception cause the test to fail?

Reply:
OOPS!!